backend/src/test/java/com/ambacam/rest/autorites/AutoritesResourceIT.java

```diff
@@ -0,0 +1,171 @@
+import com.ambacam.repository.AutoriteRepository;
+import com.ambacam.rest.ApiConstants;
+import io.restassured.http.ContentType;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+public class AutoritesResourceIT extends ItBase {
+
+	@Autowired
+	private AutoriteRepository repository;
+
+	private Autorite autorite1;
+
+	private Autorite autorite2;
+
+	@Before
+	public void setup() throws Exception {
+		super.setup();
+
+		// create autorite1
+		autorite1 = buildAutorite();
+		autorite1 = repository.save(autorite1);
+		// create autorite 2
+		autorite2 = repository.save(buildAutorite());
+	}
+
+	@Override
+	@After
+	public void cleanup() throws Exception {
+		repository.deleteAll();
+		super.cleanup();
+	}
+
+	@Test
+	public void create() {
+		Autorite create = buildAutorite();
+		int id = given().contentType(ContentType.JSON).body(create).log().body()
+				.post(ApiConstants.AUTORITE_COLLECTION).then().log().body().statusCode(200).extract().body()
+				.path("id");
+
+		// check that the autorite has been saved
+		Autorite actual = repository.findOne(Integer.toUnsignedLong(id));
+		assertThat(actual, is(notNullValue()));
+		assertThat(actual.getNom(), is(equalTo(create.getNom())));
+	}
+
+	@Test
+	public void createNomNull() {
+		Autorite create = buildAutorite();
+		create.setNom(null);
+		given().contentType(ContentType.JSON).body(create).log().body().post(ApiConstants.AUTORITE_COLLECTION)
+				.then().log().body().statusCode(400);
+	}
+
+	@Test
+	public void createNomVide() {
+		Autorite create = buildAutorite();
+		create.setNom("");
+		given().contentType(ContentType.JSON).body(create).log().body().post(ApiConstants.AUTORITE_COLLECTION)
+				.then().log().body().statusCode(400);
+	}
+
+	@Test
+	public void createPosteVide() {
+		Autorite create = buildAutorite();
+		create.setPoste("");
+		given().contentType(ContentType.JSON).body(create).log().body().post(ApiConstants.AUTORITE_COLLECTION)
+				.then().log().body().statusCode(400);
+	}
+
+	@Test
+	public void createPosteNull() {
+		Autorite create = buildAutorite();
+		create.setPoste(null);
+		given().contentType(ContentType.JSON).body(create).log().body().post(ApiConstants.AUTORITE_COLLECTION)
+				.then().log().body().statusCode(400);
+	}
+
+	@Test
+	public void createMemeNom() {
+		Autorite create = buildAutorite();
+		create.setNom(autorite1.getNom());
+		given().contentType(ContentType.JSON).body(create).log().body().post(ApiConstants.AUTORITE_COLLECTION)
+				.then().log().body().statusCode(400);
+	}
+
+	@Test
+	public void list() {
+		given().get(ApiConstants.AUTORITE_COLLECTION).then().log().body().statusCode(200)
+				.body("size()", is(equalTo(2)))
+				.body("id", containsInAnyOrder(autorite1.getId().intValue(), autorite2.getId().intValue()));
```

Comment:
Please check all properties of one autorite

Reply:
Done